File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        def dfs(city : int):
            visited.add(city)
            for neighbour , connected in enumerate(isConnected[city]):
                if connected == 1 and neighbour not in visited:
                    dfs(neighbour)
        n = len(isConnected)
        visited = set()
        provinces = 0
        for i in range(n):
            if i not in visited:
                dfs(i)
                provinces += 1
        return provinces

File: 0547-number-of-provinces/0547-number-of-provinces.c
void dfs(int city , int** isConnected , int n , bool* visited){
    visited[city] = true;
    for(int neighbour = 0; neighbour < n; neighbour++){
        if(isConnected[city][neighbour] == 1 && !visited[neighbour]){
            dfs(neighbour , isConnected , n , visited);
        }
    }
}

int findCircleNum(int** isConnected, int isConnectedSize, int* isConnectedColSize) {
    int n = isConnectedSize;
    bool visited[n];
    for(int i = 0; i < n; i++){
        visited[i] = false;
    }
    int provinces = 0;
    for(int i = 0; i < n; i++){
        if(!visited[i]){
            dfs(i , isConnected , n , visited);
            provinces++;
        }
    }
    return provinces;
}